javascript/ql/lib/semmle/javascript/frameworks/Knex.qll

```diff
@@ -57,6 +57,26 @@ module Knex {
       )
     }
 
-    override DataFlow::Node getAQueryArgument() { none() }
+    override DataFlow::Node getAResult() { 
+      (
```

Comment:
the knex model is a bit unusual in that it classifies both the expression producing the promise, as well as the `await` expression as database accesses. `getAResult`, in this case, follows the implementation of the characteristic predicate. LMK what you think.

Reply:
It seems reasonable.  
One thing you could do is to split up the class into two (one with the await, and one without). 
That might allow the code to be a bit cleaner.

Reply:
I did split it up!